pkg/webui/components/map/index.js

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import Leaflet from 'leaflet'
+
+import MarkerIcon from '../../assets/auxiliary-icons/location_pin.svg'
+
+import style from './map.styl'
+
+// Reset default marker icon
+delete Leaflet.Icon.Default.prototype._getIconUrl
+Leaflet.Icon.Default.mergeOptions({
+  iconRetinaUrl: MarkerIcon,
+  iconUrl: MarkerIcon,
+  iconSize: [26, 36],
+  shadowSize: [26, 36],
+  iconAnchor: [13, 36],
+  shadowAnchor: [8, 37],
+  popupAnchor: [0, -40],
+  shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
+})
+
+export default class Map extends React.Component {
+  getMapCenter(markers) {
+    // This will calculate zoom and map center long/lang based on all markers provided.
+    // Currently it just returns the first marker.
```

Comment:
Please link to the issue (https://github.com/TheThingsNetwork/lorawan-stack/issues/1241).
` // TODO: action (link to the issue)`

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/1129/commits/9a207f5224349c324d07fce1225aa1b2d72143bb